src/main.cpp

```diff
@@ -91,7 +91,17 @@ int main (int argc, char* argv[]) {
     MemoryParser mp(input_tracename_);
 
     // Pass to Functional Simulator
-    FunctionalSimulator fs(&rf, &stats, &mp, forward_);
+    std::unique_ptr<FunctionalSimulator> fs;
+    fs = std::make_unique<FunctionalSimulator>(&rf, &stats, &mp, forward_);
+
+    while(!fs->isProgramFinished()){
+        fs->cycle();
+
+        if(stats.getClockCycles() >= 100000) {
+            std::cerr << "Simulator did not halt within 1000 cycles";
```

Comment:
Maybe we could use a variable here to make sure the numbers are the same? Since it looks like we've got a couple orders of magnitude different :)